README.adoc

```diff
@@ -1,40 +1,96 @@
 = ALMighty UI
 
+=== ALMighty UI is a currently an application planner and issue tracker front-end.
+It at uses https://github.com/almighty/almighty-core[ALMighty-core] as the back-end.
+
 If this is the first time you are starting the UI you need to run
 
 ----
 $ npm install
 ----
 
+If you trying to refresh your install you can just run
+
+----
+$ npm run reinstall
+----
+
 Start the app by executing the following. 
 
 ----
 $ npm start
 ----
 
-If you wish to run the app with the in-memory dataset, you need to set an environment 
+If you wish to run the app with the in-memory dataset, you need to set an environment
 variable "ENV" to "inmemory" and do a rebuild. The setting is stored inside the build, 
 changing the value back to unset or "development" (default for npm start) needs a rebuild
 to take effect.
 
+== Run All Checks
+
+To run the linter, build validator, unit tests, and functional test use:
+
+----
+$ npm test
+----
+
+
+== Lint
+
+To run the TypeScript and Angular 2 linter, use:
+
+----
+$ npm run lint
+----
+
+NOTE: They also run during the unit tests.
+
+== Build Validation
+
+To validate the webpack build, use:
+
+----
+$ npm run validate
+----
+
+NOTE: Validation also runs the unit tests.
+
 == Unit Tests
 
+To run the unit test, use:
+
+----
+$ npm run test:unit
+----
+
+To run the unit test in a watch mode so that the are rerun every time you save, use:
+
+----
+$ npm watch:test
+----
+
 Unit (UI) tests are automatically executed on the PR check process (using 
```

Comment:
As this is a Readme, we could expand PR (pull request) once and then use the abbreviation for the rest of the document.